docs/eks-jobs-migration.md

```diff
@@ -0,0 +1,216 @@
+# Migrating Kubernetes Jobs To The EKS Prow Build Cluster
+
+In an ongoing effort to migrate to community-owned resources, SIG K8S Infra and
+SIG Testing are working to complete the migration of jobs from the Google-owned
+internal GKE cluster to community-owned clusters.
+
+Most of jobs in the Prow Default Build Cluster should attempt to migrate to
+`cluster: eks-prow-build-cluster`. For criteria and requirements, please
+pay attention to this document.
+
+## What is EKS Prow Build Cluster?
+
+EKS Prow Build Cluster (`eks-prow-build-cluster`) is an AWS EKS based cluster
+owned by community that's used for running ProwJobs. It's supposed to be
+similar to our GCP/GKE based clusters, but there are some significant
+differences such as:
+
+- Operating system on worker nodes where we run jobs is Ubuntu 22.04
+- The cluster is smaller in terms of number of worker nodes, but worker nodes
+  are larger (16 vCPUs, 128 GB RAM, 300 GB NVMe SSD)
+- There's cluster-autoscaler in place to scale up/down cluster on demand
+- The cluster is hosted on AWS which means we get to use the credits donation
+  that we got from AWS :)
+
+## Criteria and Requirements for Migration
+
+The following jobs can be migrated out of the box:
+
+- Jobs not using any external resources (e.g. cloud accounts)
+  - Build, lint, verify, and similar jobs
+
+The following jobs can be migrated but require some actions to be taken:
+
+- Jobs using external non-GCP resources (e.g. DigitalOcean, vSphere...)
+  - Community ownership of resources is required for jobs running in the new
+    cluster, see the Community Ownership section for more details
+
+The following jobs **MUST NOT** be migrated at this time:
+
+- Jobs using GCP resources (e.g. E2E tests, promotion jobs, etc.)
+- Jobs that are running in trusted clusters (e.g. `test-infra-trusted` and
+  `k8s-infra-prow-build-trusted`)
+
+Jobs that are already running in community-owned clusters (e.g. 
+`k8s-infra-prow-build`) can be migrated as well, but it's not required or
+mandated.
+
+## How To Migrate Jobs?
+
+Fork and check out the [kubernetes/test-infra][k-test-infra] repository,
+then follow the steps below:
+
+[k-test-infra]: https://github.com/kubernetes/test-infra
+
+- Find a job that you want to migrate
+  - You can check out the following ["Prow Results"][prow-results-default] link
+    for recent jobs that are running in the `default` cluster
+  - For explanation reasons, let's say that you picked up a job called
+    `pull-jobset-test-integration-main`
+- Edit the file that `pull-jobset-test-integration-main` is defined in. All
+  jobs are defined in the [kubernetes/test-infra][k-test-infra] repository. You
+  can use GitHub or our [Code Search tool][cs-test-infra] to find the file
+  where this job is defined (search by job name).
+- Look for a `cluster:` key in the job definition. If there isn't one or it's
```

Comment:
It might be helpful to point out that it's part of spec for cases where it's not defined.
```suggestion
- Look for a `spec.cluster` key in the job definition.
```